### PR TITLE
Validate ES min_age exporter

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Previously we hadn't checked the format of min_age, which caused
to fail the Exporter during exporting and looping in an error loop.

This PR adds a simple validation of the min_age format for the ILM configuration (ES exporter). See reference for https://www.elastic.co/guide/en/elasticsearch/reference/current/api-conventions.html#time-units

We only support days, hours, minutes, hours, and seconds. Smaller values don't make much sense. 

The check happens during the configuration, and an exception is thrown on an invalid format. This will cause the exporter director to fail. 

> [!Note]
>
> Missing is the same as the OpenSearch Exporter, but this can be seen be as a nice coding /training practice

## Related issues

relates to https://github.com/camunda/zeebe/issues/15899

<!-- Cut-off marker
_All lines under and including the cut-off marker will be removed from the merge commit message_

### CODE REVIEW

File: exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchExporter.java [hunk 1]
```diff
@@ -19,14 +19,26 @@
...
 public class ElasticsearchExporter implements Exporter {
 
+  /**
+   * Supported pattern for min_age property of ILM, we only support: days, hours, minutes and
+   * seconds. Everything below seconds we don't expect as useful.
+   *
+   * <p>See reference
+   * https://www.elastic.co/guide/en/elasticsearch/reference/current/api-conventions.html#time-units
+   */
+  private static final String PATTERN_MIN_AGE_FORMAT = "^[0-9]+[dhms]$";
```
Comment (korthout) [inline exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchExporter.java:36]: ❌ This excludes combinations of units like `3m30s`.

🔧 Perhaps there's a better way, but how about: `^([0-9]+d)?([0-9]+h)?([0-9]+m)?([0-9]+s)?$`? But you'll have to check against empty string as well.

**Review by korthout — CHANGES_REQUESTED** [review]
Thanks @Zelldon 🙇 

It looks good, but this check is a bit too restrictive. Please have a look at my suggestion

File: exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchExporter.java (hunk 1 above)
Comment (oleschoenburg) [reply exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchExporter.java:36]: According to the linked docs and https://github.com/elastic/elasticsearch/blob/293d0302fb5a32d2947e57cbfe54251c6f61a1a5/libs/core/src/main/java/org/elasticsearch/core/TimeValue.java#L338-L369 this isn't allowed anyway, just a number + unit is.

**Review by korthout — APPROVED** [review]
NVM my previous comment 😅 

LGTM 👍

**Comment by korthout** [comment]
@Zelldon Did you notice that backport to 8.1 failed? https://github.com/camunda/zeebe/pull/15928#issuecomment-1892563192

**Comment by Zelldon** [comment]
Yes I did, but had no time yet

**Comment by Zelldon** [comment]
@korthout Actually I think 8.1 doesn't make sense even 8.2 doesn't since we support ES 8 since 8.3 right?

**Comment by korthout** [comment]
@Zelldon You're right that 8.1 doesn't make sense. But we support ES 8 since 8.2:
- https://github.com/camunda/zeebe/issues/12132

**Comment by Zelldon** [comment]
Ah interesting @korthout thanks ! So I guess nothing to do here anymore :)

**Comment by korthout** [comment]
Yup, thanks for fixing it @Zelldon 🙇

**Pull request merged**